commit de788a17ed4866d96c8dbb2b01b552d5ddb43ca4 (refs/changes/35/4835/3)
Author: Mike Zeller <mike@mikezeller.net>
Date:   2018-09-13T17:55:14+00:00 (1 year, 1 month ago)
    
    OS-7243 libvarpd_c_destroy gets away with pointer murder
    Reviewed by: Dan McDonald <danmcd@joyent.com>
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Approved by: Robert Mustacchi <rm@joyent.com>

diff --git a/usr/src/lib/varpd/libvarpd/common/libvarpd_client.c b/usr/src/lib/varpd/libvarpd/common/libvarpd_client.c
index b0fa907386..260bbb6e92 100644
--- a/usr/src/lib/varpd/libvarpd/common/libvarpd_client.c
+++ b/usr/src/lib/varpd/libvarpd/common/libvarpd_client.c
@@ -10,7 +10,7 @@
  */
 
 /*
- * Copyright 2015 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
 /*
@@ -113,7 +113,7 @@ libvarpd_c_destroy(varpd_client_handle_t *chp)
 		libvarpd_panic("failed to close door fd %d: %d",
 		    client->vcl_doorfd, errno);
 
-	umem_free(chp, sizeof (varpd_client_handle_t *));
+	umem_free(chp, sizeof (varpd_client_t));
 }
 
 int
